app/javascript/controllers/audio_player_controller.js

```diff
@@ -0,0 +1,104 @@
+import { Controller } from "@hotwired/stimulus"
+import convertSecondsToDuration from "util/convertSecondsToDuration"
+
+export default class extends Controller {
+  static targets = ["audio", "duration", "progress", "progressBar"]
+
+  connect() {
+    this.playing = false
+    this.seeking = false
+    this.originalDuration = this.durationTarget.innerHTML
+    this.bindMouseMove = this.mouseMove.bind(this)
+    this.bindMouseUp = this.mouseUp.bind(this)
+  }
+
+  disconnect() {
+    window.removeEventListener("mousemove", this.bindMouseMove)
+    window.removeEventListener("mouseup", this.bindMouseUp)
+  }
+
+  async play() {
+    this.playing = true
+    if (!this.audioTarget.duration) {
+      this.setProgress(0, 1) // initial render
```

Comment:
Are the `0` and `1` acting as boolean values for time values?

Reply:
We might have figured it out:
Is it measuring progress in percentage? `0` is the beginning, and `1` represents 100%, meaning the end?

Reply:
So that method is defined as `setProgress(currentTime, totalDuration)`.  At this point, I don't actually know the duration of the audio file (the browser hasn't loaded it yet).  So I'm just arbitrarily passing 0 and 1, to set it back at 0% playback.